aldryn_client/cloud.py

```diff
@@ -85,6 +89,19 @@ def upload_boilerplate(self, archive_obj):
         )
         return request.request()
 
+    def download_db(self, website_slug):
+        request = api_requests.DownloadDBRequest(
+            self.session,
+            url_kwargs={'website_slug': website_slug},
+        )
+        temp_dir = create_temp_dir()
+        response = request.request()
+        with tarfile.open(
+                fileobj=StringIO(response.content), mode='r:gz'
```

Comment:
Do we need to wrap it into a StringIO? (it should be `BytesIO`, anyway)
I would either prefer an on-the-file extraction here if possible (it should, the tar format was created exactly to deal with this) or saving to a temporary file on disk. It's very easy to fill up the RAM with a DB dump.

(Which also raises the question: is this handled correctly on the server side? :) )